graphql/schemabuilder/types.go

```diff
@@ -44,10 +44,62 @@ var Paginated fieldFuncOptionFunc = func(m *method) {
 	m.Paginated = true
 }
 
-type TextFilterFields map[string]interface{}
+type Filter struct {
+	Name            string
+	FilterFunc      interface{}
+	BatchFilterFunc interface{}
+	FallbackFlag    func(context.Context) bool
+}
+
+func FilterFieldFunc(name string, filter interface{}) FieldFuncOption {
+	var filterStruct = Filter{
+		Name:       name,
+		FilterFunc: filter,
+	}
+
+	var fieldFuncTextFilterFields fieldFuncOptionFunc = func(m *method) {
+		if m.TextFilterFuncs == nil {
+			m.TextFilterFuncs = map[string]Filter{}
+		}
+		m.TextFilterFuncs[name] = filterStruct
+	}
+
+	return fieldFuncTextFilterFields
+}
+
+func BatchFilterField(name string, batchFilter interface{}) FieldFuncOption {
+	var filterStruct = Filter{
+		Name:            name,
+		BatchFilterFunc: batchFilter,
+	}
+
+	var fieldFuncTextFilterFields fieldFuncOptionFunc = func(m *method) {
+		if m.TextFilterFuncs == nil {
+			m.TextFilterFuncs = map[string]Filter{}
+		}
+		m.TextFilterFuncs[name] = filterStruct
+	}
+
+	return fieldFuncTextFilterFields
+}
+
+func BatchFilterFieldWithFallback(name string, batchFilter interface{}, filter interface{}, flag func(context.Context) bool) FieldFuncOption {
+	var filterStruct = Filter{
+		Name:            name,
+		FilterFunc:      filter,
+		BatchFilterFunc: batchFilter,
+		FallbackFlag:    flag,
+	}
 
-func (s TextFilterFields) apply(m *method) {
-	m.TextFilterFuncs = s
+	var fieldFuncTextFilterFields fieldFuncOptionFunc = func(m *method) {
+
+		if m.TextFilterFuncs == nil {
+			m.TextFilterFuncs = map[string]Filter{}
+		}
+		m.TextFilterFuncs[name] = filterStruct
```

Comment:
hrm, would be good to validate we're not registering the same name twice here.   Maybe add a panic?  This should only be on startup so it should be fine, and we run through this code in tests so we should crash as soon as possible.